stdlib/public/CompatibilityOverride/CompatibilityOverrideConcurrency.def

```diff
@@ -308,30 +308,10 @@ OVERRIDE_TASK_LOCAL(task_localsCopyTo, void,
                     (AsyncTask *target),
                     (target))
 
-OVERRIDE_TASK_STATUS(task_addStatusRecord, bool,
-                     SWIFT_EXPORT_FROM(swift_Concurrency), SWIFT_CC(swift),
-                     swift::, (TaskStatusRecord *newRecord), (newRecord))
-
-OVERRIDE_TASK_STATUS(task_tryAddStatusRecord, bool,
-                     SWIFT_EXPORT_FROM(swift_Concurrency), SWIFT_CC(swift),
-                     swift::, (TaskStatusRecord *newRecord), (newRecord))
-
-OVERRIDE_TASK_STATUS(task_removeStatusRecord, bool,
-                     SWIFT_EXPORT_FROM(swift_Concurrency), SWIFT_CC(swift),
-                     swift::, (TaskStatusRecord *record), (record))
-
 OVERRIDE_TASK_STATUS(task_hasTaskGroupStatusRecord, bool,
                      SWIFT_EXPORT_FROM(swift_Concurrency), SWIFT_CC(swift),
                      swift::, , )
 
-OVERRIDE_TASK_STATUS(task_attachChild, ChildTaskStatusRecord *,
-                     SWIFT_EXPORT_FROM(swift_Concurrency), SWIFT_CC(swift),
-                     swift::, (AsyncTask *child), (child))
```

Comment:
Hmm.  In the abstract, I agree that none of these should be ABI or make much sense as hook points.  However, we were considering using this one to try to fix a bug in the task-group runtime, something that's triggered from `task_create`.  We should think about the potential for that sort of thing before removing them, although maybe we can fix that particular bug without this as a hook point.

Reply:
Hmmm, `task_attachChild` is actually not called anywhere AFAICT. 

`swift_task_create_commonImpl` calls into `swift_taskGroup_attachChild` for the task group use cases and for asyncLet, it directly calls `swift_task_addStatusRecord`

Reply:
Ah, right, that's what I was thinking of.  I'm not sure what these functions are supposed to be, actually.